server/db/index.js

```diff
@@ -1,113 +1,76 @@
 const moment = require('moment-timezone');
-
-require('dotenv').config();
-const { Client } = require('pg');
-
-// clients will also use environment variables
-// for connection information
-const client = new Client({
-  user: process.env.PGUSER,
-  password: process.env.PGPASSWORD,
-  host: process.env.PGHOST,
-  database: process.env.PGDATABASE,
-  port: process.env.PORT,
-});
-
-client.connect();
-
-client.on('end', () => {
-  console.log('pg client end');
-});
-
-client.on('error', (error) => {
-  console.error('pg client error', error);
-});
-
+const restaurant = require('./models/restaurant');
 
 const bookingsToday = (restaurantId) => {
   const todayStr = moment(new Date()).tz('America/Los_Angeles').format('YYYY-MM-DD');
-  return client.query(
-    'SELECT COUNT(restaurantid) FROM reservations WHERE restaurantid=$1 AND timestamp=$2',
-    [restaurantId, todayStr],
-  );
+  return restaurant.getBookingsForDate(restaurantId, todayStr);
 };
+// console.log(bookingsToday(3000));
+
+const getMaxSeats = restaurantId => restaurant.getMaxSeats(restaurantId);
 
 const getOpenSeats = ({
   restaurantId, date,
-}) => client.query(
-  'SELECT time,(MAX(restaurants.seats)-SUM(party)) AS remaining FROM reservations INNER JOIN restaurants ON restaurants.id = reservations.restaurantid WHERE date=$1 AND restaurantid=$2 GROUP BY time',
-  [date, restaurantId],
-);
-
-
-const getMaxSeats = restaurantId => client.query(
-  'SELECT seats FROM restaurants WHERE id=$1',
-  [restaurantId],
-);
-
+}) => restaurant.getBookingsForDate(restaurantId, date)
+  .then(res => res.reduce((acc, curr) => {
+    if (acc[curr.time]) {
+      acc[curr.time] += curr.party;
+    } else {
+      acc[curr.time] = curr.party;
+    }
+    return acc;
+  }, {}))
+  .then(seatsBooked => (
+    getMaxSeats(restaurantId).then(res => (
+      Object.entries(seatsBooked).map(i => ({
+        time: Number(i[0]),
+        remaining: (res - i[1]),
+      }))
+    ))
+  ));
 
 const genReservationSlots = ({ restaurantId, date }) => Promise.all([
   bookingsToday(restaurantId),
   getOpenSeats({ restaurantId, date }),
   getMaxSeats(restaurantId),
 ])
   .then((results) => {
-    // results[0] has the # bookings made info
-    // results[1] has the timeslot & remaining seats info
-    // results[2] has the max seats for the restaurant
-    // create default reservations array with default values
-    const returnedSlots = results[1].rows.map(row => ({
-      time: row.time,
-      remaining: Number(row.remaining),
-    }));
+    const returnedSlots = results[1];
+    const returnedTimes = results[1].map(slot => slot.time);
```

Comment:
Not sure if you should use map on returnedSlots since you already set it to results[1]

Reply:
I'm using map here to make a copy of the returned slots that only contains the reservation times. That array is used to determine what times need to have a default value of the max seats remaining. I should have probably made my variable names a little more clear I'm realizing